src/security/ServerOptions.cc

```diff
@@ -244,9 +244,20 @@ Security::ServerOptions::createStaticServerContext(AnyP::PortCfg &port)
 
 #elif USE_GNUTLS
         for (auto &keys : certs) {
-            gnutls_x509_crt_t crt = keys.cert.get();
             gnutls_x509_privkey_t xkey = keys.pkey.get();
-            const auto x = gnutls_certificate_set_x509_key(t.get(), &crt, 1, xkey);
+
+            const auto certCount = 1 + keys.chain.size();
+            gnutls_x509_crt_t *crt = new gnutls_x509_crt_t[certCount];
+            crt[0] = keys.cert.get();
+            if (certCount > 1) {
+                int i = 1;
+                for (auto cert : keys.chain) {
```

Comment:
If possible:

```suggestion
                for (auto &cert : keys.chain) {
```

or even

```suggestion
                for (const auto &cert : keys.chain) {
```

Reply:
Done in 57f105b